heartbeat/processor/connectionsProcessor_test.go

```diff
@@ -0,0 +1,272 @@
+
+		cp, err := NewConnectionsProcessor(args)
+		assert.Equal(t, process.ErrNilMessenger, err)
+		assert.True(t, check.IfNil(cp))
+	})
+	t.Run("nil marshaller should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgConnectionsProcessor()
+		args.Marshaller = nil
+
+		cp, err := NewConnectionsProcessor(args)
+		assert.Equal(t, process.ErrNilMarshalizer, err)
+		assert.True(t, check.IfNil(cp))
+	})
+	t.Run("nil shard coordinator should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgConnectionsProcessor()
+		args.ShardCoordinator = nil
+
+		cp, err := NewConnectionsProcessor(args)
+		assert.Equal(t, process.ErrNilShardCoordinator, err)
+		assert.True(t, check.IfNil(cp))
+	})
+	t.Run("invalid delay should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgConnectionsProcessor()
+		args.DelayBetweenNotifications = time.Second - time.Nanosecond
+
+		cp, err := NewConnectionsProcessor(args)
+		assert.True(t, errors.Is(err, heartbeat.ErrInvalidTimeDuration))
+		assert.True(t, strings.Contains(err.Error(), "DelayBetweenNotifications"))
+		assert.True(t, check.IfNil(cp))
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		cp, err := NewConnectionsProcessor(createMockArgConnectionsProcessor())
+		assert.Nil(t, err)
+		assert.False(t, check.IfNil(cp))
+	})
+	t.Run("should work and process once", func(t *testing.T) {
+		t.Parallel()
+
+		providedConnectedPeers := []core.PeerID{"pid1", "pid2", "pid3", "pid4", "pid5", "pid6"}
+		notifiedPeers := make([]core.PeerID, 0)
+		var mutNotifiedPeers sync.RWMutex
+		args := createMockArgConnectionsProcessor()
+		expectedShard := args.ShardCoordinator.SelfId()
+		args.Messenger = &p2pmocks.MessengerStub{
+			SendToConnectedPeerCalled: func(topic string, buff []byte, peerID core.PeerID) error {
+				mutNotifiedPeers.Lock()
+				defer mutNotifiedPeers.Unlock()
+
+				shardValidatorInfo := message.ShardValidatorInfo{}
+				err := args.Marshaller.Unmarshal(shardValidatorInfo, buff)
+				assert.Nil(t, err)
+				assert.Equal(t, expectedShard, shardValidatorInfo.ShardId)
+
+				notifiedPeers = append(notifiedPeers, peerID)
+				return nil
+			},
+			ConnectedPeersCalled: func() []core.PeerID {
+				return providedConnectedPeers
+			},
+		}
+		args.DelayBetweenNotifications = 2 * time.Second
+
+		cp, _ := NewConnectionsProcessor(args)
+		assert.False(t, check.IfNil(cp))
+
+		time.Sleep(3 * time.Second)
+		_ = cp.Close()
+
+		mutNotifiedPeers.Lock()
+		defer mutNotifiedPeers.Unlock()
+
+		sort.Slice(notifiedPeers, func(i, j int) bool {
+			return notifiedPeers[i] < notifiedPeers[j]
+		})
+		assert.Equal(t, providedConnectedPeers, notifiedPeers)
+	})
+}
+
+func Test_connectionsProcessor_computeNewPeers(t *testing.T) {
+	t.Parallel()
+
+	t.Run("no peers connected", func(t *testing.T) {
+		t.Parallel()
+
+		cp, _ := NewConnectionsProcessor(createMockArgConnectionsProcessor())
+		assert.False(t, check.IfNil(cp))
+		_ = cp.Close() // avoid data races
```

Comment:
data races can happen in extreme situations: the go routine does not stop in time during high CPU usages and instead, accesses the map
In order to completely use a connection processor without the routine started, why not create a new constructor in export_test.go, something like: `NewConnectionsProcessorNoGoRoutine` 
Valid on all occurrences